diff --git a/libavformat/apetag.c b/libavformat/apetag.c
index 8464560..4724224 100644
--- a/libavformat/apetag.c
+++ b/libavformat/apetag.c
@@ -110,12 +110,17 @@ static int ape_tag_read_field(AVFormatContext *s)
 int64_t ff_ape_parse_tag(AVFormatContext *s)
 {
     AVIOContext *pb = s->pb;
-    int64_t file_size = avio_size(pb);
+    int64_t file_size;
     uint32_t val, fields, tag_bytes;
     uint8_t buf[8];
     int64_t tag_start;
     int i;
 
+    if (s->url && strstr(s->url, "://") && strncmp(s->url, "file://", 7) != 0) {
+        return 0;
+    }
+
+    file_size = avio_size(pb);
     if (file_size < APE_TAG_FOOTER_BYTES)
         return 0;
 
diff --git a/libavformat/id3v1.c b/libavformat/id3v1.c
index 3189a48..8596ef1 100644
--- a/libavformat/id3v1.c
+++ b/libavformat/id3v1.c
@@ -281,6 +281,10 @@ void ff_id3v1_read(AVFormatContext *s)
     uint8_t buf[ID3v1_TAG_SIZE];
     int64_t filesize, position = avio_tell(s->pb);
 
+    if (s->url && strstr(s->url, "://") && strncmp(s->url, "file://", 7) != 0) {
+        return;
+    }
+
     if (s->pb->seekable & AVIO_SEEKABLE_NORMAL) {
         /* XXX: change that */
         filesize = avio_size(s->pb);
